File: backend-model/src/main/java/outcobra/server/model/dto/SemesterDto.java
package outcobra.server.model.dto;

import outcobra.server.model.Semester;
import outcobra.server.model.interfaces.MappableDto;
import outcobra.server.model.interfaces.Mapper;

public class SemesterDto implements MappableDto<SemesterDto, Semester> {
    @Override
    public Mapper<SemesterDto, Semester> getMapper() {
        throw new UnsupportedOperationException("not implemented");
    }

    @Override
    public Semester toEntity() {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: backend/src/main/kotlin/outcobra/server/model/dto/emptyDto/SemesterDto.kt
package outcobra.server.model.dto.emptyDto

import outcobra.server.model.Semester
import outcobra.server.model.mapper.MappableDto
import outcobra.server.model.mapper.Mapper

class SemesterDto : MappableDto<SemesterDto, Semester> {
    override fun getMapper(): Mapper<SemesterDto, Semester> {
        throw UnsupportedOperationException("not implemented")
    }

    override fun toEntity(): Semester {
        throw UnsupportedOperationException("not implemented")
    }
//TODO Implement
}